browser/android/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -97,27 +200,36 @@ BraveCosmeticResourcesTabHelper::~BraveCosmeticResourcesTabHelper() {
 
 void BraveCosmeticResourcesTabHelper::ProcessURL(
     content::WebContents* contents,
-    content::RenderFrameHost* render_frame_host, const GURL& url) {
+    content::RenderFrameHost* render_frame_host, const GURL& url,
+    const bool& main_frame) {
   if (!ShouldDoCosmeticFiltering(contents, url)) {
     return;
   }
   g_brave_browser_process->ad_block_service()->GetTaskRunner()->
       PostTaskAndReplyWithResult(FROM_HERE,
           base::BindOnce(&GetUrlCosmeticResourcesOnTaskRunner, url.spec()),
           base::BindOnce(&GetUrlCosmeticResourcesOnUI, render_frame_host));
+  if (!main_frame)
+    return;
+  // Non-scriptlet cosmetic filters are only applied on the top-level frame
+  if (web_contents()->GetMainFrame()) {
+    web_contents()->GetMainFrame()->ExecuteJavaScriptInIsolatedWorld(
```

Comment:
Ideally we should be using extensions similar to Desktop. Given the work needed for extensions on android is large, using `Isolated Worlds` is fine. A follow-up issue to track this effort would be helpful.

Just as a defense in depth strategy, we should sanitize and validate all inputs and outputs to the JS functions would be helpful. 

@SergeyZhukovsky - Once this PR is complete, a detailed security review would be helpful.